tests/e2e/specs/gtag-events/blocks-pages.test.js

```diff
@@ -0,0 +1,271 @@
+/**
+ * External dependencies
+ */
+const { test, expect } = require( '@playwright/test' );
+
+/**
+ * Internal dependencies
+ */
+import {
+	createSimpleProduct,
+	setSettings,
+	clearSettings,
+} from '../../utils/api';
+import {
+	blockProductAddToCart,
+	relatedProductAddToCart,
+	simpleProductAddToCart,
+} from '../../utils/customer';
+import {
+	createProductCollectionBlockShopPage,
+	createProductsBlockShopPage,
+} from '../../utils/create-page';
+import { getEventData, trackGtagEvent } from '../../utils/track-event';
+
+const config = require( '../../config/default' );
+const simpleProductPrice = parseFloat( config.products.simple.regular_price );
+
+let simpleProductID;
+
+test.describe( 'GTag events on block pages', () => {
+	test.beforeAll( async () => {
+		await setSettings();
+		simpleProductID = await createSimpleProduct();
+	} );
+
+	test.afterAll( async () => {
+		await clearSettings();
+	} );
+
+	// WooCommerce shop page is built with blocks.
+	test( 'Add to cart event is sent from the shop page', async ( {
+		page,
+	} ) => {
+		const event = trackGtagEvent( page, 'add_to_cart' );
+
+		// Go to shop page (newest first)
+		await page.goto( 'shop?orderby=date' );
+		await blockProductAddToCart( page, simpleProductID );
+
+		await event.then( ( request ) => {
+			const data = getEventData( request, 'add_to_cart' );
+			expect( data.product1 ).toEqual( {
+				id: simpleProductID.toString(),
+				nm: 'Simple product',
```

Comment:
:nail_care: Maybe instead of hardcoding its name, we could read it from `config.products.simple.name`?

Reply:
I agree, but also disagree here. I personally don't like to abstract too much information in the `expect` calls. I want to see clearly what data I'm expecting without having to look it up what it should match too. It's true that if they are constants then there is no harm as they can't be modified, but this is more personal preference. Ofcourse I'm breaking my own by not using a constant for simpleProductPrice, but that's mainly because I copied that logic from GLA.

For the same reason I don't often make a lot of the expect parts in tests "DRY", which could easily be done in this case since a lot of expect similar data.

Reply:
I can relate to being explicit in `expect`s, so I'm fine leaving it as `simpleProduct*.toString()`.

But for the same reason, I'd use `config.products.simple.name` as this is what we expect from the perspective of this file. We didn't set `Simple product` here. So we expect it to be "whatever is set in config"